src/kong_backend/tests/test_add_pool.rs

```diff
@@ -0,0 +1,368 @@
+        .expect("Failed to call icrc2_allowance");
+    decode_one::<Allowance>(&response).expect("Failed to decode icrc2_allowance response")
+}
+
+const TOKEN_A_SYMBOL: &str = "TKA";
+const TOKEN_A_NAME: &str = "Test Token A";
+const TOKEN_A_FEE: u64 = 10_000;
+const TOKEN_A_DECIMALS: u8 = 8;
+
+const TOKEN_B_SYMBOL_ICP: &str = "ICP";
+const TOKEN_B_NAME_ICP: &str = "Internet Computer Protocol";
+const TOKEN_B_FEE_ICP: u64 = 10_000;
+const TOKEN_B_DECIMALS_ICP: u8 = 8;
+
+#[test]
+fn test_add_pool_setup() {
+    let (ic, kong_backend) = setup_ic_environment().expect("Failed to setup IC environment");
+
+    let controller_identity =
+        get_identity_from_pem_file(CONTROLLER_PEM_FILE).expect("Failed to get controller identity");
+    let controller_principal = controller_identity
+        .sender()
+        .expect("Failed to get controller principal");
+    let controller_account = Account {
+        owner: controller_principal,
+        subaccount: None,
+    };
+
+    let archive_options = ArchiveOptions {
+        num_blocks_to_archive: 1000,
+        max_transactions_per_response: None,
+        trigger_threshold: 500,
+        max_message_size_bytes: None,
+        cycles_for_archive_creation: None,
+        node_max_memory_size_bytes: None,
+        controller_id: controller_principal,
+        more_controller_ids: None,
+    };
+
+    let token_a_init_args = InitArgs {
+        minting_account: controller_account,
+        fee_collector_account: None,
+        transfer_fee: Nat::from(TOKEN_A_FEE),
+        decimals: Some(TOKEN_A_DECIMALS),
+        max_memo_length: Some(32),
+        token_symbol: TOKEN_A_SYMBOL.to_string(),
+        token_name: TOKEN_A_NAME.to_string(),
+        metadata: vec![],
+        initial_balances: vec![],
+        feature_flags: Some(FeatureFlags { icrc2: true }),
+        archive_options: archive_options.clone(),
+    };
+    let token_a_ledger_id = create_icrc1_ledger(
+        &ic,
+        &Some(controller_principal),
+        &LedgerArg::Init(token_a_init_args),
+    )
+    .expect("Failed to create Token A ledger");
+    let _initial_controller_balance_a = get_icrc1_balance(&ic, token_a_ledger_id, controller_account);
+
+    let token_b_principal_id = Principal::from_text("nppha-riaaa-aaaal-ajf2q-cai").expect("Invalid Testnet ICP Principal ID"); // Use Testnet ICP
+
+    let token_b_init_args = InitArgs {
+        minting_account: controller_account,
+        fee_collector_account: None,
+        transfer_fee: Nat::from(TOKEN_B_FEE_ICP),
+        decimals: Some(TOKEN_B_DECIMALS_ICP),
+        max_memo_length: Some(32),
+        token_symbol: TOKEN_B_SYMBOL_ICP.to_string(),
+        token_name: TOKEN_B_NAME_ICP.to_string(),
+        metadata: vec![],
+        initial_balances: vec![],
+        feature_flags: Some(FeatureFlags { icrc2: true }),
+        archive_options: archive_options.clone(),
+    };
+    let token_b_ledger_id = create_icrc1_ledger_with_id(
```

Comment:
for ICP, there's a separate icp_ledger as it supports accont id also. code is in common/icp_ledger.rs. So change this to create_icp_ledger(). you can add create_icp_ledger_with_id() there
```suggestion
    let token_b_ledger_id = create_icrc1_ledger_with_id(
```

Reply:
@CanerKocak   just getting use to this review suggestion. let's communicate this way